app/routes/settings/colorScheme.tsx

```diff
@@ -0,0 +1,8 @@
+import type { ActionFunction } from '@remix-run/node';
+import { redirect } from '@remix-run/node';
+import { updateUserColorScheme } from '~/models/user.server';
+
+export const action: ActionFunction = async ({ request }) => {
+    await updateUserColorScheme(request);
+    return redirect('/');
```

Comment:
nitpick: Referring to my comment above, this is where you would handle the user checking and `redirecting`